### ISSUE TITLE
Gateway startup leaks a Etcd connection

### ISSUE THREAD

**Issue by chilicat** [minio/minio#9841]
The Gateway startup leaks an Etcd connection because `lookupConfigs` is called twice. The `lookupConfigs` method sets the global variable `globalEtcdClient` which means that on the second call we have a leaked Etcd connection

See: https://github.com/minio/minio/blob/master/cmd/config-current.go#L332

The easiest way to see that it is called twice is to configure Etcd to a non-existing endpoint, it will produce an error like:

```
MINIO_ETCD_ENDPOINTS=http://localhost:2379  ./minio  gateway nas /tmp/

Error: Unable to initialize etcd config: dial tcp [::1]:2379: connect: connection refused
       2: cmd/config-current.go:334:cmd.lookupConfigs()
       1: cmd/gateway-main.go:165:cmd.StartGateway()


Error: Unable to initialize etcd config: dial tcp [::1]:2379: connect: connection refused
       5: cmd/config-current.go:334:cmd.lookupConfigs()
       4: cmd/config-current.go:589:cmd.loadConfig()
       3: cmd/config.go:250:cmd.initConfig()
       2: cmd/config.go:210:cmd.(*ConfigSys).Init()
       1: cmd/gateway-main.go:258:cmd.StartGateway()

```

**Comment by harshavardhana** [comment]
> The Gateway startup leaks an Etcd connection because `lookupConfigs` is called twice. The `lookupConfigs` method sets the global variable `globalEtcdClient` which means that on the second call we have a leaked Etcd connection

This is doesn't say anything @chilicat globalEtcdClient is only initialized if its successfully connects, if it can't nothing is initialized.

**Comment by chilicat** [comment]
In my testing the method is also called twice if you connect successfully, I just used the error case since I show nicely the trace code (which helps maybe to understand why it is called twice)

**Comment by harshavardhana** [comment]
Fixed by #9842

**Comment by chilicat** [comment]
@harshavardhana thanks for the fix.

I believe that the CoreDNS implementation leaks another Etcd connection:
https://github.com/minio/minio/blob/release/cmd/config-current.go#L358
